### DISCUSSION TITLE
Feature Request: API Explorer/Query Builder for PocketBase

### DISCUSSION THREAD

**Discussion by MarNeumann** [pocketbase/pocketbase#3277] [Q&A]
Hey guys,

I hope you're all doing well! I've been using PocketBase for a while now and have found it to be a fantastic database tool. I was wondering if there are any plans for an API explorer or query builder, similar to [GraphiQL](https://www.gatsbyjs.com/docs/how-to/querying-data/running-queries-with-graphiql/), in the near future? I think it's not on the roadmap.

I'm considering building my own tools for this, but I thought it might also be cool to have this feature in the main repository. What do you all think?

Your thoughts are much appreciated!
Marvin

**Comment by ganigeorgiev** [comment]
I can't add every user request to the roadmap - it will just never end.

Personally I don't see such much benefit of having an API explorer in the Admin UI.
I have somewhere buried in my todos to sometime in the future to consider "API rules test request" UI that will allow making requests to test the security of a collection (and save them for later runs) but for now remains a very low priority.

What would you use an API explorer for?
If you want to access and filter the collection data you already can do that from the collections listing in the Admin UI. The searchbar support both "wildcard" (search by a keyword in every field) or use the same filter syntax that is supported in the web APIs.

**Comment by MarNeumann** [comment]
Thanks for the response! I totally understand that not every user request can make it to the roadmap, and I appreciate your consideration.

The main benefit of an API explorer for me would be a quality of life improvement when working on somewhat complex queries. It makes filtering and exploring data, especially after [expanding a field](https://github.com/pocketbase/pocketbase/discussions/1160), a smoother process.

Thanks again for your time and feedback! PocketBase rocks 🔥
